### PR TITLE
Added new ROBOT_STATUS log level

### LINKED ISSUE #158: Add log support for Annunciator messages
**Is your feature request or enhancement related to a problem? What specifically could be better? Please describe.**
We need a way of logging special "Annunciator" messages so they can be shown in the Visualizer. Annunciator message are message (typically received from the robots) that report diagnostic information about the robots, such as "Robot 0 dead" or "Robot 2 Wheel 3 not commutating"

**Describe the solution you'd like**
Add a new level to the logger (added in #33) which published to a special ROS Topic that will be used by the Visualizer to display the messages.
Example, `Log(ANNUNCIATOR) << "Robot 0 low battery" << std::endl;`

**Describe alternatives you've considered**
N/A

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description
A new `ANNUNCIATOR` log level was added for the purpose of logging special messages that should be displayed in the Visualizer in a special way. Historically, Annunciator messages were diagnostic messages received from the robots over radio.

These messages are published to a special topic that the Visualizer can subscribe to, to receive and then display the messages. The messages are also logged on the `/rosout` topic with `ROS_INFO`

### Testing Done
Added additional logger ROSTests to test the `ANNUNCIATOR` messages are successfully published to both `/rosout` and the custom topic

### Resolved Issues
*Link any issues that this PR resolved. Eg `resolves #1, #2, and #5` (note that they MUST be specified like this so Github can automatically close them then this PR merges)*
resolves #158 

### Review Checklist
*(Please check every item to indicate your code complies with it (by changing `[ ]`->`[x]`). This will hopefully save both you and the reviewer(s) a lot of time!)*
***It is the reviewers responsibility to also make sure every item here has been covered***
- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom` 
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

***Feel free to make additions of things that we should be checking to this file if you think there's something missing!!!!***

### CODE REVIEW

**Comment by garethellis0** [comment]
Hmmm, if these are intended to be robot diagnostic messages, this is a great opportunity for us to use a more descriptive name. Perhaps `ROBOT_STATUS`?

**Comment by MathewMacDougall** [comment]
That's a good point. I'm not sure where the name originally came from but it was definitely only used for robot status messages, at least form the perspective of the AI.

The old GUI would show both robot status messages and dongle/radio/e-stop messages together, but that's up to the visualizer not the AI so I think `ROBOT_STATUS` is fine.

@henrymliu do you agree?

**Review by henryymliu — DISMISSED** [review]
LGTM 👍

**Comment by garethellis0** [comment]
@henrymliu - we were thinking of changing `ANNUNCIATOR` to `ROBOT_STATUS`, see above comments. Is this good with you?

**Comment by henryymliu** [comment]
Oh whoops, completely missed the discussion. I agree with the changes :+1:

**Comment by MathewMacDougall** [comment]
Updated and ready for re-review

File: src/thunderbots/software/util/logger/custom_logging_levels.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+#include <g3log/loglevels.hpp>
+
+// See https://github.com/KjellKod/g3log/blob/master/API.markdown#custom-logging-levels
+// for adding custom log levels
+
+// A custom level used to specify messages that contain robot statuses, including various
+// diagnostic information like battery voltage and wheel encoder errors. These messages
+// are separate because they are shown separately in the Visualizer.
+// over radio
+const LEVELS ROBOT_STATUS{INFO.value + 1, {"ROBOT_STATUS"}};
```
Comment (garethellis0) [inline src/thunderbots/software/util/logger/custom_logging_levels.h:12]: Is `INFO.value + 1` really the best way to do this? Seems like a bit of a hack?

**Comment by garethellis0** [comment]
Could we change the PR title so the squashed commit message gets changed?

File: src/thunderbots/software/util/logger/custom_logging_levels.h (hunk 1 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/util/logger/custom_logging_levels.h:12]: No, this is how log levels are typically added. Each log level has its own value, which affects how it's handled (eg. levels >= WARNING print to `stderr`, and levels >= FATAL will abort the system).

This values could _technically_ be the same as `INFO` and work okay, but I think this makes it clearer that the levels are distinct. This is explained in the link in the comment

**Comment by MathewMacDougall** [comment]
Done

**Review by garethellis0 — DISMISSED** [review]
LGTM :+1: :tada:

**Comment by MathewMacDougall** [comment]
Pulled from master to resolve conflicts. @garethellis0 will need re-approval

**Review by garethellis0 — DISMISSED** [review]
LGTM :+1: :tada:

**Review by henryymliu — DISMISSED** [review]

**Review by garethellis0 — APPROVED** [review]
Lgtm

**Review by henryymliu — APPROVED** [review]

**Pull request merged**